### PR TITLE
add safe check,in case virtualRenderer is undefined

### PR DESCRIPTION
**Pull request by ronniegong**
when parent component' props change and _onSizeChanged hasn't called, virtualRenderer is undefined and there throws en exception

### CODE REVIEW

File: src/recyclerlistview/RecyclerListView.js [hunk 1]
```diff
@@ -82,6 +82,9 @@ class RecyclerListView extends Component {
     }
 
     componentWillReceiveProps(newProps) {
+        if(!this._initComplete){
```
Comment (naqvitalha) [inline src/recyclerlistview/RecyclerListView.js]: What if the props have actually changed, things like layout provider etc? Simply returning here may lead to an inconsistent state

File: src/recyclerlistview/RecyclerListView.js [hunk 2]
```diff
@@ -94,6 +97,9 @@ class RecyclerListView extends Component {
 
 
     componentDidUpdate() {
+        if(!this._initComplete){
```
Comment (naqvitalha) [inline src/recyclerlistview/RecyclerListView.js]: Same new props are being used, we need to fix how virtual renderer is initialized

File: src/recyclerlistview/scrollcomponent/reactnative/ScrollComponent.js [hunk 3]
```diff
@@ -44,7 +44,7 @@ class ScrollComponent extends React.Component {
                         {...this.props}
                         horizontal={this.props.isHorizontal}
                         onScroll={this._onScroll}
-                        onLayout={(!this._isSizeChangedCalledOnce || this.props.canChangeSize) ? this._onLayout : null}>
+                        onLayout={ !this.props.containerLayout && (!this._isSizeChangedCalledOnce || this.props.canChangeSize) ? this._onLayout : null}>
```
Comment (naqvitalha) [inline src/recyclerlistview/scrollcomponent/reactnative/ScrollComponent.js]: Why is this needed?

Comment (ronniegong) [reply src/recyclerlistview/scrollcomponent/reactnative/ScrollComponent.js]: sorry,this.props.containerLayout is another feature that let user init the width and height of the container and call _onSizeChanged to finish the init at once, therefore no need to wait the onLayout call of the container

Comment (naqvitalha) [reply src/recyclerlistview/scrollcomponent/reactnative/ScrollComponent.js]: If containerlayout has been provided it should also forcefully be applied to the scroll component?

**Comment by naqvitalha** [comment]
We've fixed the null issue rightly in https://github.com/Flipkart/ReactEssentials/pull/46 .. can you raise another PR for your layout requirement?

File: src/recyclerlistview/scrollcomponent/reactnative/ScrollComponent.js (hunk 3 above)
Comment (ronniegong) [reply src/recyclerlistview/scrollcomponent/reactnative/ScrollComponent.js]: in my opinion，dont' need, but i'm not sure what 'forcefully be applied to the scroll component' mean
